File: 프로그래머스/레벨2/42746. 가장 큰 수/가장 큰 수.cpp
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

bool compare(const string &x, const string &y) {
    return (x + y) > (y + x);
}

string solution(vector<int> numbers) {
    vector<string> numbers_vector;
    for (int num : numbers) {
        numbers_vector.push_back(to_string(num));
    }

    sort(numbers_vector.begin(), numbers_vector.end(), compare);

    string result = "";
    for (const string &num : numbers_vector) {
        result += num;
    }

    return result[0] == '0' ? "0" : result;
}

File: 프로그래머스/레벨2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        List<String> numbersList = new ArrayList<>();
        for (int number : numbers) {
            numbersList.add(String.valueOf(number));
        }
        
        Collections.sort(numbersList, (x, y) -> (y + x).compareTo(x + y));
        
        String result = String.join("", numbersList);
        
        return result.charAt(0) == '0' ? "0" : result;
    }
}